src/core/stdc/config.d

```diff
@@ -208,4 +227,28 @@ else
     alias cpp_size_t = size_t;
     alias cpp_ptrdiff_t = ptrdiff_t;
 }
+
+// ABI layout of native complex types.
+private struct _Complex(T)
+{
+    T re;
+    T im;
+}
+
+version (Posix)
+{
+    align(float.alignof)  enum __c_complex_float : _Complex!float;
+    align(double.alignof) enum __c_complex_double : _Complex!double;
+    align(real.alignof)   enum __c_complex_real : _Complex!real;
+}
+else
+{
+    align(float.sizeof * 2)  enum __c_complex_float : _Complex!float;
+    align(double.sizeof * 2) enum __c_complex_double : _Complex!double;
+    align(real.alignof)      enum __c_complex_real : _Complex!real;
+}
```

Comment:
Where do the non-Posix aliases come from? - MSVC doesn't have these complex types built-in, it has `_{F,D,L}complex` typedefs (https://docs.microsoft.com/en-us/cpp/c-runtime-library/complex-math-support?view=msvc-160). `_Lcomplex` for `__c_complex_real` uses its 64-bit `long double`. The alignments are regular (4/8/8 bytes).

So is this based on the DMC runtime for 32-bit Windows?

Reply:
> So is this based on the DMC runtime for 32-bit Windows?

Or DMD 32 and 64-bit Windows.

If there's a discrepancy, the bug is in the compiler and needs fixing there.

https://github.com/dlang/dmd/blob/master/src/dmd/target.d#L214-L236

Reply:
Yeah that looks quite awful; I'll just put this behind `version(DigitalMars)` in an upcoming PR, streamlining the other definitions.